keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -232,6 +256,32 @@ message ContainerUser {
 }
 ```
 
+#### supplemental_groups_policy in RuntimeHandlerFeatures
+
+To propagate whether runtime supports fine-grained supplemental group control to `NodeRuntimeHandlerFeatures.SupplementalGroupsPolicy`, it proposes to add a corresponding field in `RuntimeHandlerFeatures`. This will also help implementing tests which interacts with CRI so that it can gracefully skip when the container runtime does not support fine-grained supplemental groups control, for example in critest.
```

Comment:
I don't know if runtime handler features is the correct level, as it's not something the OCI runtime will toggle support for, but the CRI implementation. I wonder if we need a new field in status for this?

Reply:
>  as it's not something the OCI runtime will toggle support for, but the CRI implementation

Yeah, that makes sense. Then, how about introducing `RuntimeFeatures` in `RuntimeStatus` like this?

```proto
// service RuntimeService {
//     rpc Status(StatusRequest) returns (StatusResponse) {}
// }
// message StatusResponse {
//     RuntimeStatus status = 1;
// }
message RuntimeStatus {
...
    // NEW FIELD
    RuntimeFeatures features = 2;
}
message RuntimeFeatures {
    bool supplemental_groups_policy = 1;  
}
```

> I wonder if we need a new field in status for this?

The main purpose is that I would like to merge below before container runtime releases or backports(I'm not sure this can be approved or not):
- E2E: https://github.com/kubernetes/kubernetes/pull/125470
- critest: https://github.com/kubernetes-sigs/cri-tools/pull/1438

Reply:
works for me though we'll need API review for the node status pieces. WE should make sure in the doc text of the API/CRI to make clear the distinction between the two *Features

Reply:
I'm starting to wonder if it's really safe to add this at the CRI/NodeRuntime level. Since each container is actually created by OCI runtimes, it’s something to think about.

This KEP only affects the [proc.user.additionalGids](https://github.com/opencontainers/runtime-spec/blob/main/config.md#user) field. This field is pretty important for POSIX platforms, so we can assume most OCI runtimes support it. But what if there are some OCI runtimes that don't?

Reply:
yeah LGTM!

Reply:
Addressed in 
[a998bd3](https://github.com/kubernetes/enhancements/pull/4728/commits/a998bd3a260c7a60e2d73f42c274ea1942adc0ed)

Reply:
Let me resolve this suggestion because further discussion went in [this thread](https://github.com/kubernetes/enhancements/pull/4728#discussion_r1643548866).